### PR TITLE
EI-392-display-add-bank-details-link-for-vendor-rejected

### PR DESCRIPTION
**Pull request by Stephek**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Web/Controllers/PaymentsController.cs [hunk 1]
```diff
@@ -69,7 +69,7 @@ public async Task<IActionResult> ListPaymentsForLegalEntity(string accountId, st
             {
                 Applications = submittedApplications,
                 SortField = sortField,
-                BankDetailsStatus = getApplicationsResponse.BankDetailsStatus,
+                ShowBankDetailsCalltoAction = getApplicationsResponse.BankDetailsStatus == BankDetailsStatus.NotSupplied || getApplicationsResponse.BankDetailsStatus == BankDetailsStatus.Rejected,
```
Comment (David-B-Read) [inline src/SFA.DAS.EmployerIncentives.Web/Controllers/PaymentsController.cs:72]: The bank details content panel isn't just shown for these statuses. See the EI-390 branch

File: src/SFA.DAS.EmployerIncentives.Web/ViewModels/Applications/ViewApplicationsViewModel.cs [hunk 2]
```diff
@@ -69,7 +67,7 @@ public string ToggleSortOrder(string fieldName)
             return "asc";
         }
 
-        public BankDetailsStatus BankDetailsStatus { get; set; }
+        public bool ShowBankDetailsCalltoAction { get; set; }
```
Comment (David-B-Read) [inline src/SFA.DAS.EmployerIncentives.Web/ViewModels/Applications/ViewApplicationsViewModel.cs:70]: This should be something like ShowBankDetailsRequiredCallToAction. There's different content shown for bank status of In Progress

**Review by David-B-Read — CHANGES_REQUESTED** [review]
View model refactoring should take into account all bank details statuses

File: src/SFA.DAS.EmployerIncentives.Web/Controllers/PaymentsController.cs (hunk 1 above)
Comment (Stephek) [reply src/SFA.DAS.EmployerIncentives.Web/Controllers/PaymentsController.cs:72]: In EI-390 it' shown for BankDetailsStatus.NotSupplied which is included in the code above.

File: src/SFA.DAS.EmployerIncentives.Web/ViewModels/Applications/ViewApplicationsViewModel.cs (hunk 2 above)
Comment (Stephek) [reply src/SFA.DAS.EmployerIncentives.Web/ViewModels/Applications/ViewApplicationsViewModel.cs]: OK.  Renamed to ShowAddBankDetailsCalltoAction

**Review by David-B-Read — APPROVED** [review]

**Pull request merged**